packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -148,6 +151,13 @@ export class SCWSigner implements Signer {
 
     switch (request.method) {
       case 'eth_requestAccounts': {
+        // if auto sub accounts are enabled and we have a sub account, we need to return it as a top level account
+        const subAccount = store.subAccounts.get();
+        const subAccountsConfig = store.subAccountsConfig.get();
+        if (subAccountsConfig?.enableAutoSubAccounts && subAccount?.address) {
+          this.accounts = [subAccount.address, ...this.accounts];
```

Comment:
wonder if we should put sub account at index 0 regardless of auto or dapp managed sub account?

Reply:
hmm i think that makes sense

Reply:
+1